File: Archiving/app/src/main/java/com/tokbox/sample/archiving/network/APIService.java
package com.tokbox.sample.archiving.network;

import retrofit2.Call;
import retrofit2.http.*;

public interface APIService {

    @GET("session")
    Call<GetSessionResponse> getSession();

    @POST("archive/start")
    @Headers("Content-Type: application/json")
    Call<Void> startArchive(@Body StartArchiveRequest startArchiveRequest);

    @GET("archive/{archiveId}/stop")
    Call<Void> stopArchive(@Path("archiveId") String archiveId);
}

File: Archiving-Kotlin/app/src/main/java/com/tokbox/sample/archiving/network/APIService.kt
package com.tokbox.sample.archiving.network

import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.POST
import retrofit2.http.Path

interface APIService {

    @get:GET("session")
    val session: Call<GetSessionResponse?>?

    @POST("archive/start")
    @Headers("Content-Type: application/json")
    fun startArchive(@Body startArchiveRequest: StartArchiveRequest?): Call<Any?>

    @POST("archive/{archiveId}/stop")
    fun stopArchive(@Path("archiveId") archiveId: String?): Call<Any?>
}